contract/RadicleCloudBase.sol

```diff
@@ -0,0 +1,191 @@
+    /// @notice Duration of blocks roughly equivalent to one month
+    uint32  public blocksPerMonth;
+
+    /// @notice MAXBLOCK is the maximum value of uint64
+    uint64 private constant MAXBLOCK = type(uint64).max;
+
+    /// @notice Array of when buys happened
+    uint64[] private buys    = [MAXBLOCK];
+    /// @notice Array of when expires happen
+    uint64[] private expires = [MAXBLOCK];
+    /// @notice Array of when cancels happened
+    uint64[] private cancels = [MAXBLOCK];
+    /// @notice Array of when cancel offsets happen
+    uint64[] private offsets = [MAXBLOCK];
+
+    /// @notice Tracks position of last impacted idx for all lasts
+    uint64[] private LAST = [0, 0, 0, 0, 0];
+    uint64 private constant PROCESSED   = 0;
+    uint64 private constant BUY         = 1;
+    uint64 private constant EXPIRE      = 2;
+    uint64 private constant CANCEL      = 3;
+    uint64 private constant OFFSET      = 4;
+
+    /// @notice Rate at which operator currently earns
+    uint64 private RATE = 0;
+
+    struct SubStatus {
+        // block
+        //uint64 previous;
+        // how many blocks have been purchased
+        uint128 shares;
+        // block
+        uint64 next;
+        // rate
+        uint64 consume;
+    }
+
+    /// @notice Subscription status at each block time.
+    mapping(uint64 => SubStatus) public subStatus;
+    /// @dev This is where we start our accounting from.
+    uint64 private lastProcessedSubStatus;
+    /// @dev This is the tip of statuses.
+    uint64 private lastCreatedSubStatus;
+
+    /// @notice Emitted when money is deposited for an Org, could be renewal or initial order.
+    /// @param org The Org being topped up
+    /// @param expiry Next expiry for this Org
+    event NewTopUp(address org, uint64 expiry);
+
+    // modifier to check if caller is owner
+    modifier isOwner() {
+        require(msg.sender == owner, "Caller is not owner");
+        _;
+    }
+
+    /// @dev Set contract deployer as owner.
+    constructor(uint price) {
+        owner = msg.sender;
+        pricePerMonth = price;
+        blocksPerMonth = 185_142;
+        lastProcessedSubStatus = uint64(block.number);
+        subStatus[lastProcessedSubStatus] = SubStatus({
+            shares: 0, next: 0, consume: 0
+        });
+    }
+
+    /// @dev Change owner.
+    /// @param newOwner address of new owner
+    function changeOwner(address newOwner) public isOwner {
+        owner = newOwner;
+    }
+
+    /// @dev Return owner address.
+    /// @return address of owner
+    function getOwner() external view returns (address) {
+        return owner;
+    }
+
+    /// @dev Return Org expiration block.
+    /// @return expiry of Org in block
+    function getExpiry(address org) external view returns (uint) {
+        return expiry[org];
+    }
+
+    /// @dev Top-up expiration block for an Org on payment.
+    function topUp(address org) payable public {
+        if (msg.value >= pricePerMonth) {
```

Comment:
Does this mean we can only subscribe in 1-month chunks?

Reply:
Nope, https://github.com/radicle-dev/radicle-cloud/pull/1/commits/36dcf9dff66ea8ad4d06520d100d38192e699a9f#diff-f9840e5b47321a0c9bbf85905168b75666bf3d8b6873f0cd476bbf4d3d23d2e4R105-R107